docs/src/content/docs/guides/overriding-components.md

```diff
@@ -0,0 +1,132 @@
+   This example renders a contact link.
+
+   ```astro
+   ---
+   // src/components/EmailLink.astro
+   import type { Props } from '@astrojs/starlight/props';
+   ---
+
+   <a href="mailto:houston@example.com">E-mail Me</a>
+   ```
+
+3. Tell Starlight to use your custom implementation in the [`components`](/reference/configuration/#components) configuration option in `astro.config.mjs`:
+
+   ```js {9-12}
+   // astro.config.mjs
+   import { defineConfig } from 'astro/config';
+   import starlight from '@astrojs/starlight';
+
+   export default defineConfig({
+     integrations: [
+       starlight({
+         title: 'My Docs with Overrides',
+         components: {
+           // Override the default `SocialLinks` component.
+           SocialIcons: './src/components/EmailLink.astro',
+         },
+       }),
+     ],
+   });
+   ```
+
+## Extend a built-in component
+
+You can add some UI to your site without completely replacing Starlight’s default UI by extending a built-in component.
+
+To extend a built-in component, import and render it in your custom component. In this example, we render a custom e-mail link and then also render the default `SocialLinks` component:
+
+```astro {4,8}
+---
+// src/components/EmailLink.astro
+import type { Props } from '@astrojs/starlight/props';
+import Default from '@astrojs/starlight/SocialIcons.astro';
+---
+
+<a href="mailto:houston@example.com">E-mail Me</a>
+<Default {...Astro.props}><slot /></Default>
+```
+
+When rendering a built-in component:
+
+- Spread `Astro.props` into it. This makes sure that it receives all the data it needs to render.
+- Add a `<slot />` inside the component. This makes sure that if the component is passed any child elements, Astro knows where to render them.
+
+## Use page data
+
+When overriding a Starlight component, your custom implementation receives a standard `Astro.props` object.
+This object contains all the data for the current page to help you decide what to render.
+
+For example, you can read the page’s frontmatter values as `Astro.props.entry.data`. In the following example, a replacement [`PageTitle`](/reference/overrides/#pagetitle) component uses this to display the current page’s title:
+
+```astro {5} "{title}"
+---
+// src/components/Title.astro
+import type { Props } from '@astrojs/starlight/props';
+
+const { title } = Astro.props.entry.data;
+---
+
+<h1 id="_top">{title}</h1>
+
+<style>
+  h1 {
+    font-family: 'Comic Sans';
+  }
+</style>
+```
+
+Learn more about all the available props in the [Overrides Reference](/reference/overrides/#prop-types).
+
+### Only override on specific pages
+
+Component overrides apply to all pages. However, you can use `Astro.props` to determine when to show your custom UI and when to show Starlight’s default UI.
+
+In the following example, a component overriding the [`Footer`](/reference/overrides/#footer-1) renders a custom footer on the homepage and shows the default footer on all other pages:
```

Comment:
> when I read the heading re only overriding on specific pages, I expected to see some kind of configuration option, not just conditional rendering

This was kind of the intention. One thing I anticipate is exactly this question: “OK cool, I want to to override, but only on page X” or “only on pages A, B, C”. So this was aiming to capture that user intent and then explain how to do it.

I do like the copy changes here though! Do you think it would be OK to keep the current heading and just use those? Or do you think the heading is too sneaky? 😅 

```suggestion
### Only override on specific pages

Component overrides apply to all pages. However, you can conditionally render using values from `Astro.props` to determine when to show your custom UI,  when to show Starlight’s default UI, or even when to show something entirely different.

In the following example, a component overriding Starlight's [`Footer`](/reference/overrides/#footer-1) displays "Built with Starlight 🌟" on the homepage only, and otherwise shows the default footer on all other pages:
```

Reply:
Yes would be fine! (I like front-loading the conditional rendering because again I was asking myself, "Wait, what am I missing? Isn't this just conditional rendering?" So I get that, and you get your sneaky headline!)